src/components/productDetails/productDescription.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState } from "react";
+import { connect } from "react-redux";
+import config from "../../config/config";
+
+import {
+  ProductDescriptionWrapper,
+  TaxText,
+  HorizontalLine,
+  ActionButtonWrapper,
+  ProductSubText,
+} from "./productDetailsStyledComponent";
+
+import { Row, Col, Space, Input, message, Typography, Button } from "antd";
+const { Title, Text } = Typography;
+const { Search } = Input;
+
+function ProductDescription(props) {
+  const [sizeSelected, setSizeSelected] = useState(null);
+  const { productDetails } = props;
+  const sizeArray = ["s", "m", "l"];
+
+  const product = {
+    productId: productDetails.product_id,
+    brand: productDetails.brand,
+    title: productDetails.title,
+    price: productDetails.price,
+    image: productDetails.images[0],
+    qty: 1,
+    size: sizeSelected,
+  };
+
+  const onSearch = (value) => console.log(value);
+
+  const isCartFull = (cart) => {
+    if (cart.length < config.maxCartSize) {
+      return false;
+    }
+    return true;
+  };
+
+  const getIndexOfProductInCart = (cartArray, product) => {
+    console.log(cartArray);
+    let indexOfProductInCart = null;
+    cartArray.forEach((cartProduct, index) => {
+      if (
+        cartProduct.productId === product.productId &&
+        cartProduct.size === product.size
+      ) {
+        indexOfProductInCart = index;
+      }
+    });
+    return indexOfProductInCart;
+  };
+
+  const getUpdatedCart = (cart, product) => {
+    if (isCartFull(cart)) {
+      message.error(`Cart Full`);
+      return;
+    }
+    const cartArray = [...cart];
+
+    const indexOfProductInCart = getIndexOfProductInCart(cartArray, product);
+    if (indexOfProductInCart !== null) {
+      const quantity = cartArray[indexOfProductInCart].qty++;
+      message.success(`Item Quantity increased to ${quantity + 1}`);
+    } else {
+      cartArray.push(product);
+      message.success("Item added to cart");
+    }
+    return cartArray;
+  };
+
+  const isSizeSelected = () => {
+    if (
+      productDetails.product_category === "Clothing" &&
+      sizeSelected === null
+    ) {
+      message.error("Please Select Size");
+      return false;
+    }
+    return true;
+  };
+
+  const addToCart = () => {
+    if (isSizeSelected()) {
+      const { dispatch, isCartUpdated } = props;
+      let cart = localStorage.getItem("cart");
+
+      if (!cart) {
+        localStorage.setItem("cart", JSON.stringify([product]));
+        message.success("Item added to cart");
+      } else {
+        cart = JSON.parse(cart);
+        const updatedCart = getUpdatedCart(cart, product);
+        localStorage.setItem("cart", JSON.stringify(updatedCart));
```

Comment:
define a method in utils and use it across

Reply:
used redux persist